### DISCUSSION TITLE
[Feature] Mass hiding of untagged people on an photo

### DISCUSSION THREAD

**Discussion by kingvuke** [immich-app/immich#5928] [Feature Request]
### The feature

Imagine photos from an event, in the park, group photos, etc. Immich recognize e.g. 30 people in this picture. But only 2 are important to me. I don't know any of the other people and they don't interest me.

When I look at this picture now, it would be very helpful to have a function on the right side were the faces are shown, to "hide" all untagged people in bulk.

### Platform

- [x] Server
- [X] Web
- [ ] Mobile

**Comment by Aohzan** [comment]
Hello, when we are on the `people` page, a button `Hide all unnamed people" would be great :+1:

**Comment by matticas** [comment]
Agreed;

"Hide all unnamed people" would do just that, and hopefully preserve tags of people you know in the photo; i.e. family in the photo of a crowd etc.

**Comment by chodthewacko** [comment]
I'm now trying to go through and put names on people, and I'm really getting killed by group pictures due to this.   It would be amazing to just name the 1-2 people I care about and hide the other 10-20 people in these big group pictures.

This is needed for single pictures, not just everyone on the 'people' page.   On a single photo, you know you don't care about everyone else besides 1-2 people.   On the 'people' page, there could be hundreds of people that you just haven't gotten around to naming yet.    They're both useful, but serve different needs.

**Reply by DJF3** [reply]
I actually started organizing photos in a way where big group pictures go into an 'exclude' folder because it takes hours to hide them :-)

**Comment by chodthewacko** [comment]
> I actually started organizing photos in a way where big group pictures go into an 'exclude' folder because it takes hours to hide them :-)

You can start getting kinda efficient with it.  I.e. right click each face in the group, and  'open in new tab'.  Then you can quickly hide them, it's two mouse clicks on the right to hide each face, and a control-W to close the face's tab.  The mouse clicks will all be very close to each other for the entire set, so even a large group only takes a minute or two.

**Comment by chodthewacko** [comment]
It has occurred today that something like:
right click (open in new tab), select new tab, hide, close tab  could be repeated with an autoclicker to hide everyone in a single picture.
Then you can manually unhide anyone with a name.   A bit kludgy but faster/easier than multiple clicks/buttons per person.

**Comment by mrjackwills** [comment]
I was about to open a similar feature request, I'll just copy and paste my UI idea, with the mock up.

Using this a photo [like this](https://unsplash.com/photos/aerial-photo-people-gathering-8I423fRMwjM), and the following settings, 497 faces will be found.
```json
   "facialRecognition": {
      "enabled": true,
      "modelName": "buffalo_l",
      "minScore": 0.5,
      "maxDistance": 0.5,
      "minFaces": 1
    }
```

It would be great if we could set all faces to hidden in the `edit faces` UI section. Maybe a red cross icon on the opposite corner to the blue icon, as imagined below.

![example](https://github.com/immich-app/immich/assets/32690432/a15e6622-2474-4edc-ad9e-943104591c9d)

**Comment by meokey** [comment]
just want to bump it up - I encounter the same issue, as a new Immich user, that I have to go through every and each of hundreds untagged faces to "hide" those who are strangers to me and I don't care ... 
It would be much better if the feature becomes available that with one click to hide all untagged faces from selected pictures, and I believe it could save much efforts and be much easy for people to start using Immich.

The ideal operation in my opinion is that:
click on any untagged face, Immich will show all pictures where the face shows. Select all desired pictures, and Immich should enable an menu item to "hide all untagged faces in the selected pictures". In this way, I believe, with a few clicks instead of hundreds, we can remove most strangers' faces.

**Comment by purejamie** [comment]
Hi there - I had the same problem and put together a script to do something similar, it creates an album with all the photos with minimum x number of unhidden, unamed faces which allows you to go through and name who you want. Run the script again and it hides everyone who isn't named from the photos. It's a bit hacky, but worked for me while we wait for similar functionality to come into Immich - script is here: https://github.com/purejamie/immich_scripts

**Comment by bo0tzz** [comment]
#5135

**Comment by bobtitus** [comment]
I have close to 5,000 face clusters showing in my People view — it is ridiculous not to be able to bulk edit, hide, or delete them.
The current UI requires opening each face cluster individually to hide or delete it. For users with large photo libraries — particularly photographers with thousands of portraits — this results in hundreds or thousands of face clusters that need to be managed one by one, which is completely impractical.
The same applies to photos: you cannot bulk add or edit people across multiple photos. This is a critical missing feature.
Requested features:

Bulk select in the People view — checkbox selection across multiple face clusters
Bulk actions on selection — hide, delete, merge
Filter by minimum photo count — e.g. "show only people with fewer than 5 photos" to quickly identify noise clusters
Select all filtered results — combined with the count filter, this would allow hiding all low-confidence detections in one action
Bulk add/edit people across multiple selected photos

This is standard functionality in any photo management tool that supports face recognition at scale. Without it, the People feature is effectively unusable for professional or large personal libraries.

**Comment by chodthewacko** [comment]
I actually figured out an sql command to hide all unnamed people on a photo.
I posted it to reddit here: https://www.reddit.com/r/immich/comments/1ml86n2/the_sql_way_to_hide_unnamed_faces_in_a_picture/
It's extremely annoying that it has to be done manually like that - but at least it can be done in a handful of steps.

**Reply by bobtitus** [reply]
Thanks for the SQL workaround, useful to know. That said, it is frankly ridiculous that basic face management requires direct database manipulation. The same applies across the board: no bulk hide/delete for faces, no bulk people tagging across photos, no safe folder exclusion without risking file deletion. These are day-one features for any photo management tool aimed at photographers. It feels like the developers have never managed a library of more than a few hundred photos taken by a single person. For anyone with a professional or serious amateur library, Immich's people and library management features are borderline unusable without workarounds like this one.

**Comment by Gregor1996** [comment]
I agree on all what has been said above, and I will also add another example!
Imagine being a coach of a sports team, like a running team, and having all the prize giving ceremonies. You would loave to have proper face detection that recognizes your team athletes, but you would not want that to label for you also the strong guy from another team that aways wins the races!

I have a different suggestion for solving this problem, that maybe could bring much more functionality to the Immich face detection feature set!
I propose to keep the face detection feature as is, but to have the possibility to choose a configuration that would then group faces of all people that the user manually tagged, manually gave a name to!
That would allow the user to chose which people faces to search for. Once the face detection algorithm has been run on some new faces, if no previously tagged, so grouped, faces have been found in these new images, then the user can tag a recognized face (as is done now for unassigned faces) which will let the face grouping function re-run searching for matching faces too.

I'm completely new to Immich, as I just started using it this week, but I love it!
So I could work on this feature if nobody has already started, or nobody with more skill than me could start working on this!

**Reply by Gregor1996** [reply]
@bo0tzz I just noticed that this feature request is marked as **Closed**, as duplicate!
My feature proposal goes in a different direction though, should I open a new feature request?
